python/pyspark/pandas/tests/data_type_ops/testing_utils.py

```diff
@@ -49,8 +49,15 @@ def numeric_pdf(self):
         dtypes = [np.int32, int, np.float32, float]
         sers = [pd.Series([1, 2, 3], dtype=dtype) for dtype in dtypes]
         sers.append(pd.Series([decimal.Decimal(1), decimal.Decimal(2), decimal.Decimal(3)]))
+        sers.append(pd.Series([1, 2, np.nan], dtype=float))
         pdf = pd.concat(sers, axis=1)
-        pdf.columns = [dtype.__name__ for dtype in dtypes] + ["decimal"]
+        pdf.columns = [dtype.__name__ for dtype in dtypes] + [
+            "decimal",
+            "float_nan",
+        ]
+        if LooseVersion(pd.__version__) >= LooseVersion("1.3.0"):
```

Comment:
Just to clarify, old versions of pandas fails same with/without this PR, right? Supporting it only w/ 1.3.0 is okay as it's not a regression at least.

Reply:
Yes, old versions of pandas fails same with/without this PR. This condition just to skip old versions of pandas tests, and enable all testcase for decimal('nan') with pandas v1.3.0+ version.

Reply:
Current:
- pandas on spark with old versions of pandas fails same with/without this PR. (it's very wired)
- old versions of  pandas (native pandas) support decimal.

This PR only enable the test case with new version of pandas.

Reply:
Thanks the changes look pretty good to me. Thanks for redoing this @Yikun.